app/scripts/lib/typed-message-manager.js

```diff
@@ -0,0 +1,123 @@
+const EventEmitter = require('events')
+const ObservableStore = require('obs-store')
+const createId = require('./random-id')
+const assert = require('assert')
+const sigUtil = require('eth-sig-util')
+
+
+module.exports = class TypedMessageManager extends EventEmitter {
```

Comment:
we should call this `TypedMessageController` and move it into the controller dir

Reply:
Dismissing this for now since it's non-blocking, and the current pattern matches the other message types. We can move them all together in the future.